modular_skyrat/modules/polarized_windows/capacitor.dm

```diff
@@ -0,0 +1,27 @@
+/obj/item/stock_parts/capacitor/Initialize(mapload)
+	. = ..()
+
+	RegisterSignal(src, COMSIG_ITEM_ATTACK_OBJ, PROC_REF(install_polarization_controller))
+
+
+/**
+ * Handles installing the capacitor in the window, to provide it with some
+ * polarization functionalities.
+ */
+/obj/item/stock_parts/capacitor/proc/install_polarization_controller(datum/source, obj/structure/window/target, mob/user)
+	SIGNAL_HANDLER
+
+	if(!istype(target))
+		return
+
+	. = COMPONENT_CANCEL_ATTACK_CHAIN // Just to reduce the unnecessary repetition at every early return.
+
+	var/datum/component/polarization_controller/window_polarization_controller = target.GetComponent(/datum/component/polarization_controller)
+
+	if(window_polarization_controller)
```

Comment:
Would it be possible to do this from the if statement without assigning it to a variable. I don't see the variable used anywhere else inside of the proc.

Reply:
I guess it would be, yeah.